packages/grant-explorer/src/features/common/PassportBanner.tsx

```diff
@@ -65,15 +63,64 @@ export default function PassportBanner() {
       icon: <XCircleIcon
         className="fill-pink-400 stroke-pink-100 h-7 w-7 relative text-white items-center rounded-full"/>,
       color: "bg-pink-100",
-      body: "Your Passport does not have the score needed to be eligible for donation matching.",
+      body: (
+        <div data-testid="banner-match-ineligible-body">
+          Your Passport does not have the score needed to be eligible for donation matching.
+        </div>
+      ),
       button: (
         <>
-          <button className="ml-3 font-medium text-sm underline">View score</button>
+          <button
+            className="ml-3 font-medium text-sm underline"
+            data-testid="banner-view-passport-button"
+          >
+            View score
+          </button>
           <div className="pl-1">
             <ArrowTopRightOnSquareIcon className="h-4 w-4 relative text-gray-900 items-center"/>
           </div>
         </>
       ),
+    },
+    [PassportState.LOADING]: {
+      icon: <PassportLogo className="animate-spin opacity-75"/>,
+      color: "bg-white",
+      body: (
+        <div className="opacity-50 pt-1" data-test-id="banner-loading-body">
+          Loading Passport...
+        </div>
+      ),
+      button: null,
+    },
+    [PassportState.INVALID_PASSPORT]: {
+      icon: null,
+      color: "bg-yellow-300",
+      body: (
+        <div data-testid="banner-invalid-body">
+          PLACEHOLDER
+        </div>
+      ),
+      button: (
+        <>
+          <button
+            className="ml-3 font-medium text-sm underline"
+            data-testid="banner-invalid-passport-button"
+          >
+            goto passport page thing
+          </button>
+        </>
+      )
+    },
+    [PassportState.ERROR]: {
+      icon: <ExclamationCircleIcon
+        className="fill-red-500 stroke-red-200 h-7 w-7 relative text-white items-center rounded-full"/>,
+      color: "bg-red-200",
+      body: (
+        <div data-testid="banner-error-connected-body">
+          An error occurred while loading your Passport. Please try again later.
+        </div>
+      ),
+      button: null,
```

Comment:
This seems to getting harder to read! 
Could we do writing them as isolated functions for the sake of readability?

Reply:
Funny - I really prefer this! I edited the code to be simpler by adding a test id to its config. This makes sense because the element's differences are defined by its state so we can only focus on how they differ.

my 2c xD